google/cloud/storage/internal/noex_client.h

```diff
@@ -98,7 +99,173 @@ class Client {
     return raw_client_;
   }
 
+  //@{
+  /**
+   * @name Pub/Sub operations.
+   *
+   * Cloud Pub/Sub Notifications sends information about changes to objects in
+   * your buckets to Cloud Pub/Sub, where the information is added to a Cloud
+   * Pub/Sub topic of your choice in the form of messages.
+   *
+   * @see https://cloud.google.com/storage/docs/pubsub-notifications for more
+   *     information about Cloud Pub/Sub in the context of GCS.
+   */
+  /**
+   * Retrieves the list of Notifications for a Bucket.
+   *
+   * Cloud Pub/Sub Notifications sends information about changes to objects
+   * in your buckets to Google Cloud Pub/Sub service.
+   *
+   * @param bucket_name the name of the bucket.
+   * @param options a list of optional query parameters and/or request headers.
+   *     Valid types for this operation include `UserProject`.
+   *
+   * @par Idempotency
+   * This is a read-only operation and is always idempotent.
+   *
+   * @par Example
+   * @snippet storage_notification_samples.cc list notifications
+   */
+  template <typename... Options>
+  StatusOr<std::vector<NotificationMetadata>> ListNotifications(
+      std::string const& bucket_name, Options&&... options) {
+    internal::ListNotificationsRequest request(bucket_name);
+    request.set_multiple_options(std::forward<Options>(options)...);
+    auto result = raw_client_->ListNotifications(request);
+    if (not result.first.ok()) {
+      return std::move(result.first);
```

Comment:
There should be no need to `return std::move(x)` here (or in general, I think). The compiler is required to do the efficient RVO or move automatically if it can. So if our code doesn't generate the efficient code, then we may be missing a needed constructor in our StatusOr<T> implementation. But for now, we may still want to just `return result.first;`, because that's how we'd want the calling code to be written.

Reply:
Fixed.

Reply:
Now fixed for realsies. But also: https://godbolt.org/z/FBGQH7

Reply:
Thanks. For posterity, my original comment and reasoning was flawed. `return local_identifier;` should not have a std::move. But `return foo.bar;` *should* have a std::move, exactly as Carlos had originally written it. Thanks for digging in to godbolt to demonstrate this.